src/xp.runner/commands/List.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Linq;
+using System.Collections.Generic;
+using Xp.Runners.IO;
+using Xp.Runners.Config;
+using Xp.Runners.Exec;
+
+namespace Xp.Runners.Commands
+{
+    /// <summary>The list command lists available subcommands</summary>
+    public class List : Command
+    {
+        private IEnumerable<Type> BuiltinsIn(Assembly assembly)
+        {
+            return assembly
+                .GetTypes()
+                .Where(t => t.Namespace == "Xp.Runners.Commands")
+                .Where(t => t.IsSubclassOf(typeof(Command)))
+            ;
+        }
+
+        private IEnumerable<EntryPoint> ScriptsIn(string dir)
+        {
+            return Directory
+                .GetFiles(dir, "xp.*")
+                .Where(f => !f.EndsWith(".bat"))
+                .Select(f => new EntryPoint(f))
```

Comment:
On Windows, both `script` and `script.bat` files are created (the first is intended for use w/ Cygwin).